ports/unix/mpconfigport.h

```diff
@@ -154,8 +154,12 @@ typedef long mp_off_t;
 // Don't default sys.argv and sys.path because we do that in main.
 #define MICROPY_PY_SYS_PATH_ARGV_DEFAULTS (0)
 
+#if defined(__wasi__)
```

Comment:
Should be done like 

```
#ifndef  MICROPY_PY_SYS_EXECUTABLE
#define MICROPY_PY_SYS_EXECUTABLE (1)
#endif
```

so it works generically, then make a separate mpconfigvariant_wasi.h where it gets set to t 0.

Reply:
who would include mpconfigvariant_wasi.h?

Reply:
That would be done automatically if you use `make VARIANT=wasi` (though, my mistake, it should actually be unix/variants/wasi/mpconfigvariant.h). Likewise instead of build.sh you'd implement the specific makefile options in unix/variants/wasi/mpconfigvariant.mk). Check out the other variants for examples. Also I think to have any chance of merging this ulab should probably not be included unconditionally because this makes it harder to build out of the box; on the other hand if this only works with non-approved upstream wasi patches it's not super useful for others anyway.

Reply:
Yes, please make this a new variant.  It should then be possible to build it via `make VARIANT=wasi`.

Reply:
ok. i was not aware of the mechanism. thank you. let me try to adapt.

Reply:
> Also I think to have any chance of merging this ulab should probably not be included unconditionally because this makes it harder to build out of the box; on the other hand if this only works with non-approved upstream wasi patches it's not super useful for others anyway.

sure.

wrt wasi patches, alternatively it's possible to put the setjmp/longjmp runtime within micropython itself. at least until when setjmp/longjmp gets ready in wasi. how do you think?

anyway, it won't be super useful until wasm EH is widely implemented in runtimes though.

Reply:
If building unix for wasi requires patches to wasi/binaryen then IMO this PR should stay as a draft until those patches are no longer needed.  You could put the setjmp/longjmp code here, but I guess the other patches are still needed.

Reply:
i adapted to use VARIANT. it simplify a lot. thank you.

i'm not sure what should be done for UNAME_S logic in Makefile though.
a build for WASI is actually a cross compile. but the existing CROSS_COMPILE seems geared to "arm-none-eabi-gcc" style cross toolchain, which doesn't seem suitable to wasi-sdk style cross toolchain.

Reply:
> If building unix for wasi requires patches to wasi/binaryen then IMO this PR should stay as a draft until those patches are no longer needed. You could put the setjmp/longjmp code here, but I guess the other patches are still needed.

FYI, all toolchain patches have been merged in the upstream since then.